cache/src/commonMain/kotlin/org/mobilenativefoundation/store/cache5/Identifiable.kt

```diff
@@ -0,0 +1,5 @@
+package org.mobilenativefoundation.store.cache5
+
+interface Identifiable<Id : Any> {
```

Comment:
what does this wrapping get us?

Reply:
It enables caching of individual items from a collection. The collection is associated with a key. But the item-level keys need to be deterministic so a user can load the correct item. For instance, we can't generate each item-level key by appending metadata. The main alternative I considered is a key generator the user could provide. I think that's more flexible but also adds overhead. Thoughts?